go/libraries/doltcore/sqle/testutil.go

```diff
@@ -165,6 +165,7 @@ func drainIter(iter sql.RowIter) error {
 		if err == io.EOF {
 			break
 		} else if err != nil {
+			_ = iter.Close()
```

Comment:
Probably prefer to just defer this at the beginning of the function

Reply:
And actually, we aren't capturing the error from Close(), if any, which is a serious bug that has caused problems in the past. Definitely needs to be fixed.

Reply:
This is only used in tests, we can just panic here instead.